File: 2015/01/aoc2015_01.pl
#!/usr/bin/env perl6
my $accum = 0, my $i = 0, my $negind = Nil;
while defined $_ = $*IN.getc {
	given $_ {
		when '(' { $accum += 1; }
		when ')' { $accum -= 1; }
		default { next }
	};
	$i++;
	$negind = $i if $accum < 0 && !defined $negind
}
say $accum, ' ', $negind
# vim: set noet sw=4 ts=4:

File: 2015/01/aoc2015_01.c
#include <stdio.h>

int main() {
	int accum = 0, c, i = 0, negind = -1;
	while((c = getchar()) != EOF) {
		if(c == '(') accum += 1;
		else if(c == ')') accum -= 1;
		if(++i && (accum < 0) && (negind == -1)) negind = i;
	}
	printf("%d %d\n", accum, negind);
	return 0;
}
